aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -143,50 +143,82 @@ static void check_1d(const Tensor& t, const char* arg, const char* fn) {
 }
 
 Tensor addr(const Tensor& self, const Tensor& vec1, const Tensor& vec2, Scalar beta, Scalar alpha) {
-  check_1d(vec1, "vec1", "addr");
-  check_1d(vec2, "vec2", "addr");
-  Tensor b_self;
-  std::tie(b_self) = expand_size(self, {vec1.size(0), vec2.size(0)}, "addr");
-  return at::_addr(b_self, vec1, vec2, beta, alpha);
+  Tensor result = at::empty({0}, self.options());
```

Comment:
This won't participate in type promotion properly. Make sure you add a test for this behavior.

The easiest thing to do here is probably just to redundantly implement addr and not call addr_out:

```
outer_result = torch.outer(vec1, vec2).mul_(alpha);

if (beta.to<double>() != 0.0) {
  return outer_result.add_(self * beta);
}

return outer_result;
```

Reply:
Update: let's also deprecate `addr` by throwing a warning here, too, with the same language as suggested for its documentation. (This is a more aggressive deprecation plan than we previously discussed, sorry for the change.)